modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java

```diff
@@ -0,0 +1,203 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.pc.core.executors;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.governance.registry.extensions.interfaces.Execution;
+import org.wso2.carbon.pc.core.ProcessCenterConstants;
+import org.wso2.carbon.registry.core.jdbc.handlers.RequestContext;
+import org.wso2.carbon.registry.core.session.CurrentSession;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.registry.resource.services.utils.AddRolePermissionUtil;
+
+import java.util.Map;
+
+/**
+ * This executor can be used to change role permissions of a process when changing the states of the lifecycle.
+ */
+public class LCExecutor implements Execution {
+
+    private static final Log log = LogFactory.getLog(LCExecutor.class);
+    private String user = null;
+    private String path = null;
+    private UserRegistry userRegistry = null;
+
+    public void init(Map map) {
+    }
+
+    /**
+     * Implementing the execute method
+     *
+     * @param currentState
+     * @param targetState
+     * @return
+     */
+    public boolean execute(RequestContext context, String currentState, String targetState) {
+
+        boolean updated;
+
+        try {
+            user = context.getResource().getAuthorUserName();
+            String[] roles = CurrentSession.getUserRealm().getUserStoreManager().getRoleNames();
+            path = context.getResourcePath().toString();
+            userRegistry = (UserRegistry) context.getSystemRegistry();
+
+            setPublisherPermission(currentState, targetState);
+            for (String role : roles) {
+                if (role.equals("admin") || role.equals("Internal/publisher") || role
+                        .equals("Internal/private_" + user)) {
+                    continue;
+                } else {
+                    setOtherRolePermission(role, currentState, targetState);
+                }
+            }
+            setPrivateRolePermission(user, currentState, targetState);
+
+            updated = true;
+
+        } catch (Exception e) {
+            String errMsg = "Failed to update Permission";
+            log.error(errMsg, e);
+            throw new Error(errMsg, e);
+        }
+
+        return updated;
+    }
+
+    /**
+     * This method is used to identify the permissions to be assigned and denied from the permission string and add or
+     * remove the permission.
+     *
+     * @param role
+     * @param permissionString
```

Comment:
Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7